server/routes/public/forum_routes.py

```diff
@@ -29,12 +29,16 @@ async def get_posts(
     session: SessionDep,
     subject_id: int,
     user_id: int | None = None,
-    filter_tags: List[int] = Query(None)
+    filter_tags: List[int] = Query(None),
+    search_keyword: str | None = None
 ) -> list[ForumPostResponse]:
-    """Get all posts by tags, also gets all posts reaction information based on the user_id"""
+    """Get all posts by tags, also gets all posts reaction information based on the user_id
+        If have a search keyword in the request, return posts with that word
```

Comment:
questão de inglês tbm: "If a search keyword is present in the request" acho que fica melhor